packages/cli/src/reporters/github.ts

```diff
@@ -0,0 +1,129 @@
+import * as fs from 'fs'
+import * as path from 'path'
+
+import AbstractListReporter, { checkFilesMap } from './abstract-list'
+import { formatDuration, printLn, getTestSessionUrl, getTraceUrl } from './util'
+
+const outputFile = './checkly-github-report.md'
+
+type GithubMdBuilderOptions = {
+  testSessionId?: string
+  testResultIds?: { [key: string]: string }
+  numChecks: number
+  runLocation: string
+  checkFilesMap: checkFilesMap
+}
+
+export class GithubMdBuilder {
+  testSessionId?: string
+  testResultIds?: { [key: string]: string }
+  numChecks: number
+  runLocation: string
+  checkFilesMap: checkFilesMap
+  subHeader: Array<string>
+  tableHeaders: Array<string>
+  extraTableHeadersWithLinks: Array<string>
+  tableRows: Array<string> = []
+
+  readonly header: string = '# Checkly Test Session Summary'
+  readonly tableSeparatorFiller: string = '|:-'
+  readonly tableSeparator: string = '|'
+  constructor (options: GithubMdBuilderOptions) {
+    this.testSessionId = options.testSessionId
+    this.testResultIds = options.testResultIds
+    this.numChecks = options.numChecks
+    this.runLocation = options.runLocation
+    this.checkFilesMap = options.checkFilesMap
+
+    this.subHeader = []
+    this.tableHeaders = ['Result', 'Name', 'Check Type', 'Filename', 'Duration']
+    this.extraTableHeadersWithLinks = ['Assets', 'Link']
+    this.tableRows = []
+  }
+
+  render (): string {
+    this.subHeader.push(`Ran **${this.numChecks}** checks in **${this.runLocation}**.`)
+
+    if (this.testSessionId) {
+      this.subHeader.push(`[View detailed test session summary](${getTestSessionUrl(this.testSessionId)})`)
+    }
+
+    if (this.testSessionId && this.testResultIds) {
+      this.tableHeaders = this.tableHeaders.concat(this.extraTableHeadersWithLinks)
+    }
+
+    for (const [_, checkMap] of this.checkFilesMap.entries()) {
+      for (const [_, { check, result }] of checkMap.entries()) {
+        const tableRow: Array<string> = [
+          `${result.hasFailures ? '❌ Fail' : '✅ Pass'}`,
+          `${result.name}`,
+          `${result.checkType}`,
+          `\`${result.sourceFile}\``,
+          `${formatDuration(result.responseTime)} `,
+        ]
+
+        if (this.testSessionId && this.testResultIds) {
+          const assets: Array<string> = []
+
+          if (result.hasFailures && result.traceFilesUrls) {
+            assets.push(`[Trace](${getTraceUrl(result.traceFilesUrls[0])})`)
+          }
+
+          if (result.hasFailures && result.videoFilesUrls) {
+            assets.push(`[Video](${result.videoFilesUrls[0]})`)
+          }
+
+          const assetsColumn: string = assets.join(' \\| ')
```

Comment:
@tnolet , same as above, maybe we can remove the spaces to avoid multiple line in the cell content for small width table.